From 1e1ca47ec137235d55ebe48352941e9f74c80f6c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 25 Sep 2014 13:59:25 +0100
Subject: [PATCH 18/24] window: Always set workspace state while constructing

set_workspace_state () returns early when the desired sticky state
and workspace match the current property values, assuming that the
corresponding MRU lists are already correct in that case.
However that might not be the case when we are setting the initial
state, so don't take the shortcut in that case.

https://bugzilla.gnome.org/show_bug.cgi?id=737178
---
 src/core/window.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/core/window.c b/src/core/window.c
index faf0aa9..f46ca08 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -4311,7 +4311,8 @@ set_workspace_state (MetaWindow    *window,
     g_return_if_fail ((window->constructing && on_all_workspaces) || window->unmanaging);
 
   if (on_all_workspaces == window->on_all_workspaces &&
-      workspace == window->workspace)
+      workspace == window->workspace &&
+      !window->constructing)
     return;
 
   if (window->workspace)
-- 
1.8.4

